firmware/main.c

```diff
@@ -555,8 +555,13 @@ static void cmd_motor_rt(BaseSequentialStream *chp, int argc, char *argv[]){
 
         }
 
-        if (!dOpt || !pOpt || !mOpt)
-                goto exit_with_usage;
+        if (!mOpt
+		|| ((strcmp(mOpt, "1") == 0) && (!pOpt || !dOpt))
+		|| ((strcmp(mOpt, "2") == 0) && (!pOpt || !dOpt)) 
+		|| ((strcmp(mOpt, "3") == 0) && !sOpt)
+		|| ((strcmp(mOpt, "4") == 0) && !sOpt)
+|| !((strcmp(mOpt, "1") == 0) || (strcmp(mOpt, "2") == 0) || (strcmp(mOpt, "3") == 0) || (strcmp(mOpt, "4") == 0))
```

Comment:
The structure here looks a little funky. Long line, and closing paren right in front of the goto?